activities/Measure.activity/js/activity.js

```diff
@@ -0,0 +1,438 @@
+						document.getElementById("timeratevalue").addEventListener('click',function(){
+							if(retreived_datastore.time_flag==1){
+								p.typ=this.value;
+								retreived_datastore.timeyp=p.typ;
+								document.getElementById("timeratevalue").value=retreived_datastore.timeyp;
+							}
+							
+						});
+						p.timegraphtitle=webL10n.get("TimeGraph")
+						p.xaxis1=p.createP(p.timegraphtitle);
+						p.xaxis1.style('padding-left:40%');
+						p.xaxis1.style('color:white');
+					};
+
+					p.draw = function(){
+						p.background(p.bg);
+						let waveform = p.fft.waveform();
+						p.noFill();
+						p.beginShape();
+						p.stroke(255);
+						for (let i = 0; i < waveform.length; i++){
+							let x = p.map(i*p.txp, 0, waveform.length, 0, p.width);
+							let y = p.map( waveform[i]*p.typ, -1, 1, 0, p.height);
+							p.vertex(x,y);
+						}
+						p.endShape();
+					};
+					p.windowResized = function(){
+						p.resizeCanvas(window.innerWidth-100,window.innerHeight-200);
+						document.getElementById("fullscreen-button").addEventListener('click',function(){
+							p.resizeCanvas(window.innerWidth-50,window.innerHeight-100);
+						});
+						document.getElementById("unfullscreen-button").addEventListener('click',function(){
+							p.resizeCanvas(window.innerWidth-100,window.innerHeight-200);
+						});
+					};
+				};
+			var myp5 = new p5(s,'one');
+		
+			document.getElementById("timebased").style.visibility="hidden";
+			document.getElementById("timebased").style.display="none";
+			document.getElementById("timebased").style.width="0px";
+			document.getElementById("timebased").style.height="0px";
+			document.getElementById("freqbased").style.visibility="visible";
+			document.getElementById("freqbased").style.width="47px";
+			document.getElementById("freqbased").style.height="47px";
+			}
+		};
+
+		document.getElementById("timebased").addEventListener("click",function(event){
+			//test for data.palettechecker=1
+			if(data.palettechecker==1){
+				retreived_datastore.freq_flag=0;
+				retreived_datastore.time_flag=1;
+				timebased(retreived_datastore);
+				document.getElementById("one").style.display="block";
+				document.getElementById("two").style.display="none";
+				document.getElementById("freqbased2").style.width="47px";
+				document.getElementById("freqbased2").style.height="47px";
+				document.getElementById("freqbased2").style.left="-20px";
+				document.getElementById("freqbased2").style.visibility="visible";
+				retreived_datastore.last_graph=1;
+			}
+			else{
+				data.freq_flag=0;
+				data.time_flag=1;
+				timebased(check_data);
+				data.last_graph=1;
+			}
+			
+		});
+	
+		function freqbased(argument_values,x){
+			retreived_datastore.last_graph=0;
+			if(document.getElementById("freqbased").value == 0){
+				var t = function(p){
+					p.mic2;
+					p.cnv;
+					p.yslider;
+					p.xslider;
+					p.yaxis;
+					p.xaxis;
+					p.bg;
+					p.px;
+					p.py;
```

Comment:
Not sure to understand what this code do?

Reply:
It was for testing purpose. Fixed